worlds/Files.py

```diff
@@ -112,14 +113,25 @@ def get_manifest(self) -> Dict[str, Any]:
         }
 
 
-class APDeltaPatch(APContainer, metaclass=AutoPatchRegister):
-    """An APContainer that additionally has delta.bsdiff4
+class APPatch(APContainer, abc.ABC, metaclass=AutoPatchRegister):
+    """
+    An abstract `APContainer` that defines the requirements for an object
+    to be used by the `Patch.create_rom_file` function.
+    """
+    result_file_ending: str = ".sfc"
```

Comment:
I think defaulting to ".sfc" here isn't good

Reply:
I wondered about that, but I was just keeping the current behavior.
Changing it might require changes to worlds, and could break apworlds that rely on it.
(since this is inherited into `APDeltaPatch`)

Reply:
Note that declaring it more abstractly here, and then defining the ".sfc" default in `APDeltaPatch`, would run into the same mypy bug that we're dealing with for `Option.default`.
(not that we should make all our design decisions around a mypy bug, but just a small factor to consider)

Reply:
After letting it sink in a bit, the extra layer of abstraction is probably a good idea (for now at least) and the .sfc argument is understandable.
Kinda would like @Silvris to chime in/approve.

Reply:
For the ".sfc", changing it would involve adjusting several worlds (at minimum, checked Lufia 2 and LttP and both rely on this behavior) so it's worth keeping if just to not increase the complexity of this PR.

The abstraction is worthwhile and provides additional benefits while APPP (as one dev already had to swap to APPP pre-maturely due to issues with bsdiff crashing).